scripts/generate-redirects-from-migrations/generate-redirect-utils.go

```diff
@@ -0,0 +1,237 @@
+	// Query the Renku API to get the list of migrated projects
+	body := queryRenkuApi(renkuCredentials, "/renku_v1_projects/migrations", nil, queryApiFlags{})
+	if body == nil {
+		return nil
+	}
+
+	type migrationResponse struct {
+		ProjectID  string `json:"project_id"`
+		V1ID       uint   `json:"v1_id"`
+		LauncherID string `json:"launcher_id"`
+	}
+
+	var migrations []migrationResponse
+	if err := json.Unmarshal(body, &migrations); err != nil {
+		fmt.Printf("Error parsing JSON response: %v\n", err)
+		return nil
+	}
+
+	projects := make([]MigratedProject, 0, len(migrations))
+	for _, m := range migrations {
+		projects = append(projects, MigratedProject{
+			v1Id:      m.V1ID,
+			projectId: m.ProjectID,
+		})
+	}
+	return projects
+}
+
+func HasExistingRedirect(renkuCredentials ServerCredentials, projectRedirectDesc ProjectRedirectDescription) bool {
+	sourceUrl := fmt.Sprintf("/projects/%s", projectRedirectDesc.gitLabProjectPathWithNamespace)
+	endpoint := fmt.Sprintf("/platform/redirects/%s", url.PathEscape(sourceUrl))
+	body := queryRenkuApi(renkuCredentials, endpoint, nil, queryApiFlags{accept404: true})
+	if body == nil {
+		return false
+	}
+
+	var config PlatformRedirectConfig
+	if err := json.Unmarshal(body, &config); err != nil {
+		fmt.Printf("Error parsing platform redirect config: %v\n", err)
+		return false
+	}
+	return true
+}
+
+func RetrieveGitLabProjectIdentifier(gitlabCredentials ServerCredentials, migratedProject MigratedProject) string {
+	// Query the GitLab API to get the project identifier
+	url := fmt.Sprintf("https://%s/api/v4/projects/%d", gitlabCredentials.host, migratedProject.v1Id)
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		fmt.Printf("Error creating GitLab request: %v\n", err)
+		return ""
+	}
+	req.Header.Set("PRIVATE-TOKEN", gitlabCredentials.token)
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		fmt.Printf("Error calling GitLab API: %v\n", err)
+		return ""
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		fmt.Printf("GitLab API returned status: %d\n", resp.StatusCode)
+		return ""
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		fmt.Printf("Error reading GitLab response: %v\n", err)
+		return ""
+	}
+
+	var result struct {
+		PathWithNamespace string `json:"path_with_namespace"`
+	}
+	if err := json.Unmarshal(body, &result); err != nil {
+		fmt.Printf("Error parsing GitLab JSON: %v\n", err)
+		return ""
+	}
+
+	return result.PathWithNamespace
+}
+
+func ScanCredentials(name string) ServerCredentials {
+	var token string
+	hostEnvVar := fmt.Sprintf("%s_HOST", strings.ToUpper(name))
+	if envHost := os.Getenv(hostEnvVar); envHost != "" {
+		fmt.Printf("Using %s host from environment variable (%s): %s\n", name, hostEnvVar, envHost)
+		defaultHosts[name] = envHost
+	} else {
+		fmt.Printf("No %s host environment variable (%s) found, using default: %s\n", name, hostEnvVar, defaultHosts[name])
+	}
+	host := defaultHosts[name]
```

Comment:
not the end of the world, but it seems strange to insert the custom value in the defaults map so that you can use it later...

Also it could be that we wont be able to find a default value for the host.


```suggestion
	hostEnvVar := fmt.Sprintf("%s_HOST", strings.ToUpper(name))
	var host string
	if envHost := os.Getenv(hostEnvVar); envHost != "" {
		fmt.Printf("Using %s host from environment variable (%s): %s\n", name, hostEnvVar, envHost)
		host = envHost
	} else {
		fmt.Printf("No %s host environment variable (%s) found, using default: %s\n", name, hostEnvVar, defaultHosts[name])
		var hostFound bool
		host, hostFound = defaultHosts[name]
		if !hostFound {
			fmt.Printf("Could not find default host for %s, exiting.\n", name)
			os.Exit(1)
		}
	}
	
```

Reply:
Fixed in 7e2eb71f6cd5df8fffd4e9e2905722a20dd91bb4